contracts/core/BorrowableMarginPool.sol

```diff
@@ -0,0 +1,288 @@
+            oTokenBalance.mul(_amount).div(collateralAllocatedToMM)
+        );
+        // transfer _amount of collateralAsset from pool to borrower
+        ERC20Interface(collateralAsset).safeTransfer(msg.sender, _amount);
+        emit Borrow(_oToken, collateralAsset, _amount, msg.sender);
+    }
+
+    /**
+     * @notice get the amount borrowable by market maker
+     * @param _borrower address of the borrower
+     * @param _oToken address of the oToken laying claim to the collateral assets
+     */
+    function borrowable(address _borrower, address _oToken) external view returns (uint256) {
+        OtokenInterface oToken = OtokenInterface(_oToken);
+        address collateralAsset = oToken.collateralAsset();
+
+        uint256 collateralAssetDecimals = ERC20Interface(collateralAsset).decimals();
+
+        uint256 collateralAllocatedToMM = ERC20Interface(_oToken).balanceOf(_borrower);
+        if (collateralAssetDecimals >= 8) {
+            collateralAllocatedToMM = collateralAllocatedToMM.mul(10**(collateralAssetDecimals.sub(8)));
+        } else {
+            collateralAllocatedToMM = collateralAllocatedToMM.div(10**(uint256(8).sub(collateralAssetDecimals)));
+        }
+
+        uint256 modifiedBal = collateralAllocatedToMM.mul(borrowPCT[collateralAsset]).div(TOTAL_PCT);
+        return
+            whitelistedBorrower[_borrower] &&
+                !oToken.isPut() &&
+                oToken.expiryTimestamp() > block.timestamp &&
+                modifiedBal > borrowed[_borrower][collateralAsset]
+                ? modifiedBal.sub(borrowed[_borrower][collateralAsset])
+                : 0;
+    }
+
+    /**
+     * @notice Repays asset back to pool before oToken expiry
+     * @param _oToken address of the oToken laying claim to the collateral assets
+     * @param _amount amount of the asset to repay
+     */
+    function repay(address _oToken, uint256 _amount) public {
+        _repay(_oToken, _amount, msg.sender, msg.sender);
+    }
+
+    /**
+     * @notice Repays asset back to pool for another borrower before oToken expiry
+     * @param _oToken address of the oToken laying claim to the collateral assets
+     * @param _amount amount of the asset to repay
+     * @param _borrower address of the borrower to repay for
+     */
+    function repayFor(
+        address _oToken,
+        uint256 _amount,
+        address _borrower
+    ) public {
+        require(_borrower != address(0), "MarginPool: Borrower cannot be zero address");
+        _repay(_oToken, _amount, _borrower, msg.sender);
+    }
+
+    /**
+     * @notice Repays asset back to pool for another borrower before oToken expiry
+     * @param _oToken address of the oToken laying claim to the collateral assets
+     * @param _amount amount of the asset to repay
+     * @param _borrower address of the borrower to repay for
+     * @param _repayer address of the repayer of the loan
+     */
+    function _repay(
+        address _oToken,
+        uint256 _amount,
```

Comment:
`_amount` is confusing between oToken amount of which amount, IMO better to name it `_collateralAmount`